### ISSUE TITLE
Big conv2d fails optimizer validation

### ISSUE THREAD

**Issue by ppadjinTT** [tenstorrent/tt-mlir#6537]
### Describe the bug

When trying to pass a larger convolution through the optimizer with optimization level 1, this error appears:

`<unknown>:0: error: loc("convolution.5"): OperationValidationAndFallback: Operation ttnn.conv2d failed validation (original error: OutOfMemoryError - Not enough L1 memory). No fallback configuration worked (tested up to 10000 combinations).` This error **does not appear** when optimization level is 0.

This is the configuration of the given convolution:
- input: (2, 960, 128, 128) bf16
- weight: (320, 960, 3, 3) bf16
- padding: 1
- stride: 1
- dilation: 1
- groups: 1

## From TTIR
Run the attached repro.mlir file with following command: 
```
TTMLIR_RUNTIME_LOGGER_LEVEL=Trace TTRT_LOGGER_LEVEL=Debug  ./build/bin/ttmlir-opt --ttir-to-ttnn-backend-pipeline="system-desc-path=$SYSTEM_DESC_PATH optimization-level=1" --mlir-print-ir-after-all  -o out.mlir repro.mlir &> repro.log
```

## From tt-xla
```
import torch
import torch.nn as nn
import torch_xla
import torch_xla.core.xla_model as xm
import torch_xla.runtime as xr


def run_conv2d_repro():
    xr.set_device_type("TT")

    # Match optimization level from sdxl-pipeline.py
    torch_xla.set_custom_compile_options({"optimization_level": 1})

    # Create conv2d with exact parameters from failing SDXL op
    model = nn.Conv2d(
        in_channels=960,
        out_channels=320,
        kernel_size=3,
        stride=1,
        padding=1,
        dilation=1,
        groups=1,
        bias=False,  # No bias in the failing op
    )
    model = model.to(torch.bfloat16)
    model = model.eval()

    # Create input with exact shape from failing op
    # input: tensor<2x960x128x128xbf16>
    input_tensor = torch.randn(2, 960, 128, 128, dtype=torch.bfloat16)

    # Get torch reference output (on CPU)
    print(f"Input shape: {input_tensor.shape}")
    print(f"Weight shape: {model.weight.shape}")
    print(f"Computing torch reference output...")
    with torch.no_grad():
        torch_output = model(input_tensor)
    print(f"Output shape: {torch_output.shape}")

    # Compile with TT backend
    print(f"Compiling model with backend='tt'...")
    model = torch.compile(model, backend="tt")

    # Move to device and run
    device = xm.xla_device()
    input_device = input_tensor.to(device)
    model = model.to(device)

    print(f"Running on TT device...")
    with torch.no_grad():
        tt_output = model(input_device)

    tt_output_cpu = tt_output.cpu()
    print(f"TT output shape: {tt_output_cpu.shape}")
    print(f"TT output sample: {tt_output_cpu[0, 0, 0, :5]}")


if __name__ == "__main__":
    run_conv2d_repro()
```

**Comment by ppadjinTT** [comment]
@odjuricicTT can you take a look when you find time?

**Comment by rpavlovicTT** [comment]
Can you attach TTIR mlir file for reproduction?

**Comment by ppadjinTT** [comment]
oops, didn't see i didn't attach. here it is (renamed it to .txt since .mlir is not supported for upload ) 

[repro.txt](https://github.com/user-attachments/files/24584916/repro.txt)

**Comment by rpavlovicTT** [comment]
@bmalesevicTT please take a look at this. It's worth trying making slice config fallback to dram, just like we have fallback for conv2d config today.

**Comment by ppadjinTT** [comment]
without optimizer this operation passes (i assume it just fallbacks on DRAM). So doing nothing to the original conv config will probably suffice

**Comment by ppadjinTT** [comment]
Solved by https://github.com/tenstorrent/tt-mlir/pull/6569
